File: setup-logo.py
#!/usr/bin/env python3
"""
Quick setup: Save your logo image as 'logo-source.png' in the same folder as this script,
then run: python setup-logo.py
"""

import os
import sys
from pathlib import Path

# Check if Pillow is available
try:
    from PIL import Image
    HAS_PIL = True
except ImportError:
    HAS_PIL = False

def setup_logo():
    """Convert and optimize logo for web"""
    
    source_logo = Path("logo-source.png")
    output_dir = Path("frontend/public")
    output_dir.mkdir(parents=True, exist_ok=True)
    
    if not source_logo.exists():
        print("❌ Please save your logo as 'logo-source.png' first")
        print("   Run this script again once the file is in place.")
        return False
    
    if not HAS_PIL:
        print("⚠️  Pillow not installed. Install with: pip install pillow")
        return False
    
    try:
        print("📦 Processing logo...")
        img = Image.open(source_logo)
        
        # Generate favicon (32x32 ICO)
        print("  → Creating favicon.ico (32x32)")
        if img.mode == 'RGBA':
            bg = Image.new('RGB', img.size, (255, 255, 255))
            bg.paste(img, mask=img.split()[3])
            img_ico = bg
        else:
            img_ico = img.convert('RGB')
        
        img_ico_32 = img_ico.resize((32, 32), Image.Resampling.LANCZOS)
        img_ico_32.save(output_dir / 'favicon.ico', format='ICO')
        
        # Main logo PNG (keep original size or max 512px)
        print("  → Creating logo.png")
        max_size = 512
        if img.width > max_size or img.height > max_size:
            img.thumbnail((max_size, max_size), Image.Resampling.LANCZOS)
        img.save(output_dir / 'logo.png', 'PNG', optimize=True)
        
        # Generate sizes for different uses
        sizes = {
            'logo-192x192.png': 192,  # PWA
            'logo-256x256.png': 256,  # Social preview
            'logo-512x512.png': 512,  # PWA large
        }
        
        for filename, size in sizes.items():
            print(f"  → Creating {filename}")
            img_resized = Image.open(source_logo)
            img_resized.thumbnail((size, size), Image.Resampling.LANCZOS)
            img_resized.save(output_dir / filename, 'PNG', optimize=True)
        
        # WebP version (better compression)
        print("  → Creating logo.webp")
        img_webp = Image.open(source_logo)
        img_webp.save(output_dir / 'logo.webp', 'WEBP', quality=90)
        
        print(f"\n✅ All done! Your logo files are ready in {output_dir}/")
        print("\n📁 Generated files:")
        for f in ['favicon.ico', 'logo.png', 'logo-192x192.png', 'logo-256x256.png', 
                  'logo-512x512.png', 'logo.webp']:
            file_path = output_dir / f
            if file_path.exists():
                size_kb = file_path.stat().st_size / 1024
                print(f"  ✓ {f:25} ({size_kb:.1f} KB)")
        
        return True
        
    except Exception as e:
        print(f"❌ Error: {e}")
        return False

if __name__ == "__main__":
    success = setup_logo()
    sys.exit(0 if success else 1)

File: setup-logo.js
#!/usr/bin/env node

/**
 * Logo setup script for State House Boys Senior School
 * This script converts your logo image to multiple formats for web use
 * 
 * Usage: node setup-logo.js
 */

const fs = require('fs');
const path = require('path');

console.log('🎓 State House Boys Senior School - Logo Setup\n');

// Check for required dependencies
let sharp;
try {
  sharp = require('sharp');
} catch (e) {
  console.log('📦 Installing sharp (image processor)...\n');
  require('child_process').execSync('npm install sharp', { stdio: 'inherit' });
  sharp = require('sharp');
}

const sourceFile = path.join(process.cwd(), 'logo-source.png');
const outputDir = path.join(process.cwd(), 'frontend', 'public');

async function setupLogo() {
  // Check if source file exists
  if (!fs.existsSync(sourceFile)) {
    console.error('❌ Error: logo-source.png not found');
    console.log('\n📋 Please follow these steps:');
    console.log('   1. Download the school logo image you provided');
    console.log('   2. Save it as "logo-source.png" in the project root:');
    console.log('      ' + sourceFile);
    console.log('   3. Run this script again: node setup-logo.js\n');
    process.exit(1);
  }

  // Ensure output directory exists
  if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
  }

  try {
    console.log('📦 Processing logo...\n');

    // Read source image
    const sourceImage = sharp(sourceFile);
    const metadata = await sourceImage.metadata();
    console.log(`   Source: ${metadata.width}×${metadata.height} pixels\n`);

    // Generate favicon (32x32 ICO)
    console.log('  → Generating favicon.ico (32×32)');
    await sharp(sourceFile)
      .resize(32, 32, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 1 } })
      .toFormat('png')
      .toFile(path.join(outputDir, 'favicon-temp.png'));
    
    // Convert PNG to ICO using raw buffer (simpler approach)
    const pngBuffer = await sharp(sourceFile)
      .resize(32, 32, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 1 } })
      .toFormat('png')
      .toBuffer();
    
    fs.writeFileSync(path.join(outputDir, 'favicon.png'), pngBuffer);
    console.log('     ✓ favicon.png (use .ico converter online if needed)');

    // Main logo PNG
    console.log('  → Generating logo.png');
    const maxSize = 512;
    const width = Math.min(metadata.width, maxSize);
    const height = Math.min(metadata.height, maxSize);
    
    await sharp(sourceFile)
      .resize(width, height, { fit: 'inside', withoutEnlargement: true })
      .toFormat('png')
      .toFile(path.join(outputDir, 'logo.png'));
    console.log('     ✓ logo.png');

    // Generate sizes for different uses
    const sizes = [
      { name: 'logo-192x192.png', size: 192 },
      { name: 'logo-256x256.png', size: 256 },
      { name: 'logo-512x512.png', size: 512 },
    ];

    for (const { name, size } of sizes) {
      console.log(`  → Generating ${name}`);
      await sharp(sourceFile)
        .resize(size, size, { fit: 'contain', background: { r: 255, g: 255, b: 255, alpha: 0 } })
        .toFormat('png')
        .toFile(path.join(outputDir, name));
      console.log(`     ✓ ${name}`);
    }

    // Generate WebP version
    console.log('  → Generating logo.webp');
    await sharp(sourceFile)
      .toFormat('webp')
      .toFile(path.join(outputDir, 'logo.webp'));
    console.log('     ✓ logo.webp');

    console.log('\n✅ All done! Your logo files are ready:\n');
    console.log(`   📁 ${outputDir}/\n`);

    // List generated files
    const files = [
      'favicon.png',
      'logo.png',
      'logo-192x192.png',
      'logo-256x256.png',
      'logo-512x512.png',
      'logo.webp',
    ];

    for (const file of files) {
      const filePath = path.join(outputDir, file);
      if (fs.existsSync(filePath)) {
        const sizeKb = (fs.statSync(filePath).size / 1024).toFixed(1);
        console.log(`   ✓ ${file.padEnd(25)} (${sizeKb} KB)`);
      }
    }

    console.log('\n💡 Tip: favicon.png works, but for IE11 compatibility:');
    console.log('   Convert favicon.png to .ico format at: https://icoconvert.com/\n');

  } catch (error) {
    console.error('❌ Error processing logo:', error.message);
    process.exit(1);
  }
}

setupLogo();
